tests/integration_tests/functional/test_mmds.py

```diff
@@ -122,6 +122,70 @@ def test_custom_ipv4_support_config(test_microvm_with_ssh, network_config):
     )
 
 
+def test_json_response(test_microvm_with_ssh, network_config):
+    """Test the API for MMDS custom ipv4 support."""
+    test_microvm = test_microvm_with_ssh
+    test_microvm.spawn()
+
+    response = test_microvm.mmds.get()
+    assert test_microvm.api_session.is_status_ok(response.status_code)
+    assert response.json() == {}
+
+    data_store = {
+        'latest': {
+            'meta-data': {
+                'ami-id': 'ami-12345678',
+                'reservation-id': 'r-fea54097',
+                'local-hostname': 'ip-10-251-50-12.ec2.internal',
+                'public-hostname': 'ec2-203-0-113-25.compute-1.amazonaws.com',
+            }
+        }
+    }
+    response = test_microvm.mmds.put(json=data_store)
+    assert test_microvm.api_session.is_status_no_content(response.status_code)
+
+    response = test_microvm.mmds.get()
+    assert test_microvm.api_session.is_status_ok(response.status_code)
+    assert response.json() == data_store
+
+    config_data = {
+        'ipv4_address': '169.254.169.250',
+        'json_response': True
+    }
+    response = test_microvm.mmds.put_config(json=config_data)
+    assert test_microvm.api_session.is_status_no_content(response.status_code)
+
+    test_microvm.basic_config(vcpu_count=1)
+    _tap = test_microvm.ssh_network_config(
+         network_config,
+         '1',
+         allow_mmds_requests=True
+    )
+
+    test_microvm.start()
+    ssh_connection = net_tools.SSHConnection(test_microvm.ssh_config)
+
+    response = test_microvm.mmds.put_config(json=config_data)
+    assert test_microvm.api_session.is_status_bad_request(response.status_code)
+
+    cmd = 'ip route add 169.254.169.250 dev eth0'
+    _, stdout, stderr = ssh_connection.execute_command(cmd)
+    _assert_out(stdout, stderr, '')
+
+    pre = 'curl -s http://169.254.169.250/'
+
+    cmd = pre + 'latest'
+    _, stdout, stderr = ssh_connection.execute_command(cmd)
+    _assert_out(stdout, stderr, '{"meta-data":{"ami-id":"ami-12345678",\
```

Comment:
Should we make sure that the output is a valid JSON as well?

Reply:
I think we are already covered through these checks, which confront the response content with an expected valid JSON. Do you want me to add further JSON validation checks on the expected JSON, to make sure we do not invalidate the expected JSON when modifying these tests?